src/option_parser.cr

```diff
@@ -199,10 +216,19 @@ class OptionParser
   end
 
   private struct ParseTask
-    @double_dash_index : Int32?
+    @stop_index : Int32?
+
+    def initialize(@parser : OptionParser, @args : Array(String), order = false)
+      double_dash_index = @stop_index = @args.index("--")
+
+      # stop at first nondash param unless "--" comes earlier
+      if order && (order_index = @args.index { |arg| arg =~ /^[^\-]/ })
+        unless (ddi = double_dash_index) && ddi < order_index
```

Comment:
shouldn't need the `ddi = double_dash_index` here, since `double_dash_index` is a local variable.